### PR TITLE
bugfix: Handle more edge cases in from_html!

### PR DESCRIPTION
**Pull request by pygarap**
> [!IMPORTANT]
> Closing #886  

This pull request improves the robustness and accuracy of HTML parsing and validation in the `air.tags` module. The main changes include replacing the old HTML detection logic with a more comprehensive approach, restructuring the parsing methods for better clarity and correctness, and adding new tests to cover edge cases such as leading whitespace and invalid HTML.

**HTML Detection and Validation Improvements:**

* Introduced a new regular expression `_LOOKS_LIKE_HTML_UNICODE_RE` in `constants.py` to better identify HTML-like strings.
* Added the `looks_like_html` utility function in `utils.py`, combining `nh3.is_html` detection and the new regex for more reliable HTML validation.
* Updated `from_html` in `BaseTag` to use `looks_like_html` instead of `nh3.is_html`, providing stricter validation and improved error messaging for invalid HTML input.

**Parsing Logic Refactoring:**

* Refactored the recursive parsing logic in `BaseTag` by introducing `_from_lexbor_node` and `_from_element_node` methods, clarifying the distinction between element nodes and other node types. [[1]](diffhunk://#diff-0de0eb99c9c8181ae6a7fd339542b6aaf46c643ef916083fe9beba79dc61fa33L719-R736) [[2]](diffhunk://#diff-0de0eb99c9c8181ae6a7fd339542b6aaf46c643ef916083fe9beba79dc61fa33L757-R769)
* Added `_is_lexbor_html_parser_valid` in `utils.py` to encapsulate parser validation logic, ensuring required HTML structure is present before proceeding.

**Testing Enhancements:**

* Added new tests to `test_base_tag.py` for handling HTML with leading whitespace, standalone comments, and various invalid HTML cases, ensuring the new logic behaves as expected.

**Dependency and Import Updates:**

* Moved `nh3` import from `models/base.py` to `utils.py` and updated import statements to reflect new utilities and constants usage. [[1]](diffhunk://#diff-0de0eb99c9c8181ae6a7fd339542b6aaf46c643ef916083fe9beba79dc61fa33L11) [[2]](diffhunk://#diff-f2079f9aa1dd8d0ce2760ffd2bd3c15bdb84ff0892b16f8f154ca3ea08e4a331R17) [[3]](diffhunk://#diff-f2079f9aa1dd8d0ce2760ffd2bd3c15bdb84ff0892b16f8f154ca3ea08e4a331R37) [[4]](diffhunk://#diff-0de0eb99c9c8181ae6a7fd339542b6aaf46c643ef916083fe9beba79dc61fa33R27) [[5]](diffhunk://#diff-0de0eb99c9c8181ae6a7fd339542b6aaf46c643ef916083fe9beba79dc61fa33R41) [[6]](diffhunk://#diff-f69eb5ec528f35f5bd27e812e54b2327d8c1f307eb01a2a565779eb13bc4bef5L10-R10)

These changes collectively make HTML parsing in the `air.tags` module more robust, maintainable, and better tested.

### CODE REVIEW

File: src/air/tags/constants.py [hunk 1]
```diff
@@ -115,3 +115,21 @@ class TagKeys:
     """,
     re.IGNORECASE | re.DOTALL | re.VERBOSE,
 )
+_LOOKS_LIKE_HTML_UNICODE_RE: Final = re.compile(
```
Comment (pydanny) [inline src/air/tags/constants.py:118]: Any chance this could be targeted with an XML bomb attack?

I know this is for converting HTML to Tags, but still I wonder about the safety of this.

File: tests/test_base_tag.py [hunk 2]
```diff
@@ -19,13 +19,16 @@
     TINY_AIR_TAG_SOURCE_SAMPLE,
 )
 from examples.samples.html_samples import FRAGMENT_HTML_SAMPLE, HTML_SAMPLE, SMALL_HTML_SAMPLE, TINY_HTML_SAMPLE
+from full_match import match as full_match
```
Comment (pydanny) [inline tests/test_base_tag.py:22]: Fascinating library, just saw what it does. 🙏🏻

File: src/air/tags/constants.py (hunk 1 above)
Comment (pygarap) [reply src/air/tags/constants.py:118]: @pydanny Thanks for raising this.

> [!IMPORTANT]
> This regex runs only once as a Boolean gate check!

This code path is not an XML parser and it does not perform any entity resolution or expansion, so an “XML bomb” style attack does not apply here. We are only doing two Boolean gate checks to decide whether an input string should be treated as “looks like HTML”:

* `nh3.is_html(text)` is a broad heuristic check for whether markup is present.
* `_LOOKS_LIKE_HTML_UNICODE_RE.fullmatch(text)` is a stricter shape check for the inputs we accept.

Also, the regex is pre-compiled once at import time and then reused, so we are not repeatedly compiling patterns on user input, and we are not building any DOM/tree or transforming content at this stage.

The only realistic risk category here is not XML bombs but potential regex worst case backtracking (ReDoS) on very large adversarial inputs.

**Review by pydanny — APPROVED** [review]
